src/structure/pdb.jl

```diff
@@ -30,26 +46,314 @@ end
+obsolete PDB directory;
+if the keyword argument `obsolete_dir` is set the obsolete PDB files are 
+downloaded to the specified directory;
+"""
+function downloadallobsoletepdb(;obsolete_dir::AbstractString=pwd(), file_format::Type=PDB)
+    # Get all obsolete PDB files in RCSB PDB Server using getallobsolete() and download them
+    obsoletelist = pdbobsoletelist()
+    downloadpdb(obsoletelist, pdb_dir=obsolete_dir, file_format=file_format)
+end
+
+
+"""
+Download and read the PDB file or biological assembly from the RCSB PDB server.
+By default downloads the PDB file to current working directory; 
+if the keyword argument `ba_number` is set the biological assembly with that 
+number will be downloaded; 
+if the keyword argument `pdb_dir` is set the PDB file is downloaded to the 
+specified directory; 
+if the keyword argument `obsolete` is set `true`, the PDB file is downloaded 
+into the obsolete directory inside `pdb_dir`;
+if the keyword argument `overwrite` is set `true`, then it will overwrite the
+PDB file if it exists in the `pdb_dir`;
+"""
+function retrievepdb(pdbid::AbstractString;
+            pdb_dir::AbstractString=pwd(),
+            obsolete::Bool=false, 
+            overwrite::Bool=false, 
+            ba_number::Integer=0,
+            structure_name::AbstractString="$pdbid.pdb",
+            kwargs...)
+    downloadpdb(pdbid, pdb_dir=pdb_dir, obsolete=obsolete, overwrite=overwrite, ba_number=ba_number)
+    info("Parsing the PDB file...")
+    if obsolete
+        # if obsolete is set true, the PDB file will downloaded to obsolete directory inside "pdb_dir"
+        filepath = joinpath(pdb_dir,"obsolete","$pdbid.pdb")
+    else    
+        filepath = joinpath(pdb_dir,"$pdbid.pdb")
+    end
+    pdbpath = ba_number == 0 ? filepath : filepath*"$ba_number"
+    open(pdbpath, "r") do input
+        read(input, PDB; structure_name=structure_name, kwargs...)
+    end
+end
+
+"""
+Read the PDB file based on PDB ID and PDB directory specified.
+if the keyword argument `pdb_dir` is set the PDB file is read from the 
+specified directory; 
+Keyword argument `structure_name` is used for representing the PDB structure
+when parsing the file. 
+if the keyword argument `remove_disorder` is set true, then disordered atoms 
+wont be parsed
+if the keyword argument `read_std_atoms` is set false, then standard 
+ATOM records wont be parsed
+if the keyword argument `read_het_atoms` is set false, then hetro 
+HETATOM records wont be parsed
+"""
+function readpdb(pdbid::AbstractString;
+            pdb_dir::AbstractString=pwd(),
+            ba_number::Integer=0,
+            structure_name::AbstractString="$pdbid.pdb",
+            remove_disorder::Bool=false,
+            read_std_atoms::Bool=true,
+            read_het_atoms::Bool=true)
+    filepath = joinpath(pdb_dir,"$pdbid.pdb")
+    pdbpath = ba_number == 0 ? filepath : filepath*"$ba_number"
+    open(pdbpath, "r") do input
```

Comment:
This `open` is not actually required as calling `read` with the filepath is already defined below.